rust-runtime/aws-smithy-http-client/src/client/tls.rs

```diff
@@ -181,3 +181,55 @@ cfg_rustls! {
         }
     }
 }
+
+cfg_s2n_tls! {
+    /// s2n-tls based support and adapters
+    pub mod s2n_tls_provider {
```

Comment:
The only reason the rustls mod is public is really for the `CryptoMode` enum. If we don't need to expose anything here then maybe make it `pub(crate)` instead.